src/cpp/hu.bme.mit.inf.modes3.ESP32-sensor/railsensor_3/ESP32-sensor/lib/SensorStatemachine/SensorStatemachine.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef SENSOR_STATEMACHINE_HPP
+#define SENSOR_STATEMACHINE_HPP
+
+const double maxtime = 4;
+
+enum StateMachine
+{
+  Trainwait,
+  Firstdetect,
+  Traindetect,
+  Nextwait,
+  Datasend
+};
+
+class SensorStateMachine
+{
+private:
+  bool detect;
+  int kocsiszam;
+  bool direction;
+  StateMachine state;
+
+public:
+  InfSensor sensor[2];
```

Comment:
Same review as above for public fields:

`Also, should this field be public? Public fields always imply a gut feeling, because in this way anyone with malicious intensions, may modify this field, without the object (who encapsulates it) knowing it.`